docs/designs/supporting_pytorch.md

```diff
@@ -0,0 +1,297 @@
+# Design doc about Supporting PyTorch
+
+## introduction
+ElasticDL is an open-source distributed deep learning programming framework based on TensorFlow 2.x and Kubernetes. At the Google Developer Day event in the fall of 2019, the ElasticDL team from Ant Financial showed off the first open-source version of ElasticDL.The primary design intent of ElasticDL is to simplify distributed programming. It allows users to provide only models described with TensorFlow 2.0 API, without requiring users to write distributed training process code. As long as the user's model definition can be adjusted locally, the model can be trained with large-scale data in a distributed environment, thereby improving R&D efficiency.
+
+However, this project only supports tensorflow. Considering that PyTorch is more widely used in academia, this project will improve and support PyTorch.
+
+## Dataloader
+PyTorch has an abstract Dataset class. A Dataset can be anything that has a `__len__` function (called by Python’s standard `len` function) and a `__getitem__` function as a way of indexing into it. This[tutorial](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html) walks through a nice example of creating a custom `FacialLandmarkDataset` class as a subclass of `Dataset`.
+
+PyTorch’s [TensorDataset](https://pytorch.org/docs/stable/_modules/torch/utils/data/dataset.html#TensorDataset) is a Dataset wrapping tensors. By defining a length and way of indexing, this also gives us a way to iterate, index, and slice along the first dimension of a tensor. This will make it easier to access both the independent and dependent variables in the same line as we train.
+
+DataLoader is a tool used to package your data. So you have to load your own (NumPy array or other) data format into Tensor, and then put it into this wrapper. Using DataLoader help you iterate data efficiently.
```

Comment:
"to package your data" maybe not accurate. The description of `DataLoader` on  is

`torch.utils.data.DataLoader` is an iterator that provides all these features. 
- Batching the data
- Shuffling the data
- Load the data in parallel using multiprocessing workers.